source/_integrations/ecodevices.markdown

```diff
@@ -0,0 +1,124 @@
+---
+title: GCE Eco-Devices
+description: Instructions on how to integrate GCE Eco-Devices within Home Assistant.
+ha_release: 0.89
+ha_iot_class: Local Polling
+ha_category:
+  - Sensor
+ha_codeowners:
+  - '@Aohzan'
+ha_config_flow: true
+ha_domain: ecodevices
+---
+
+The `ecodevices` integration allows you to get information from [GCE Eco-Devices](http://gce-electronics.com/fr/carte-relais-ethernet-module-rail-din/409-teleinformation-ethernet-ecodevices.html).
+
+## Configuration
+
+To add ecodevices to your installation, go to Configuration >> Integrations in the UI, click the button with + sign and from the list of integrations select GCE Eco-Devices.
+
+Alternatively, you need to add the following to your configuration.yaml file:
+
+```yaml
+# Example configuration.yaml entry
+sensor:
+  - platform: ecodevices
+    host: "192.168.1.239"
+    t1_enabled: True
+    t1_name: Compteur Linky
+    c1_enabled: True
+    c1_name: Compteur panneau solaire
+    c1_icon: mdi:solar-panel
+    c1_unit_of_measurement: Wh
+    c1_device_class: power
```

Comment:
New integrations, that communicate with a device or service, should not provide an option for YAML configuration.

Please remove this functionality, as it is only intended for transitional/migration purposes of existing integrations.

Reply:
Ok, I will update it to delete the YAML part and provide only configuration from GUI.

For "complexe" integration (like https://github.com/Aohzan/ipx800#example, that I will subit soon too) which need a lot of parameter. The YAML isn't possible too ?

Reply:
Exceptions are possible, for example, MQTT/KNX, which define a complex protocol/specification that isn't possible with the UI at this point.

However, in this PR; that is not the case, as a config flow is provided already.

Reply:
yes of course, so I have to think about the ipx800v4 integration